x/feemarket/ante/fee.go

```diff
@@ -41,96 +45,109 @@ func (dfd FeeMarketCheckDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		return ctx, sdkerrors.ErrInvalidGasLimit.Wrapf("must provide positive gas")
 	}
 
-	minGasPrices, err := dfd.feemarketKeeper.GetMinGasPrices(ctx)
+	requiredBaseFee, err := dfd.feemarketKeeper.GetMinGasPrice(ctx)
 	if err != nil {
 		return ctx, errorsmod.Wrapf(err, "unable to get fee market state")
 	}
 
-	fee := feeTx.GetFee()
+	feeCoins := feeTx.GetFee()
 	gas := feeTx.GetGas() // use provided gas limit
 
+	if len(feeCoins) > 1 {
+		return ctx, feemarkettypes.ErrTooManyFeeCoins
+	}
+
+	// If there is a fee attached to the tx, make sure the fee denom is a denom accepted by the chain
+	if len(feeCoins) == 1 {
+		feeDenom := feeCoins.GetDenomByIndex(0)
+		if feeDenom != requiredBaseFee.Denom {
+			return ctx, err
+		}
+	}
```

Comment:
maybe im missing something, but if the fee denom isnt the same should the resolver be used? it seems like if the base fee is something like eth and i try paying with usdc, it wouldnt ever hit the resolver.

Reply:
what does `GetDenomByIndex` do?

Reply:
Yeah I wasn't sure if the `DenomResolver` implementation should handle that or if we should.

I'll just add the check.

Reply:
this is supposed to return an error or nil?

Reply:
im thinking we probably should run it through the resolver just so that we can set the appropriate priority on the context.